source/common/http/async_client_impl.h

```diff
@@ -233,7 +248,13 @@ class AsyncStreamImpl : public virtual AsyncClient::Stream,
   void setDecoderBufferLimit(uint64_t) override {
     IS_ENVOY_BUG("decoder buffer limits should not be overridden on async streams.");
   }
-  uint64_t decoderBufferLimit() override { return buffer_limit_.value_or(0); }
+  uint64_t decoderBufferLimit() override {
+    if (new_async_client_retry_logic_) {
+      return buffer_limit_.value_or(kDefaultDecoderBufferLimit);
+    } else {
+      return buffer_limit_.value_or(0);
+    }
```

Comment:
Now, we have a default 64K buffer limit. So, for the stream request, we won't buffer too much data if no explicit buffer_limit is set.